### ISSUE TITLE
Many out of range responses when spinning pokestops.

### ISSUE THREAD

**Issue by mewio** [Noctem/Monocle#275]
Running scan.py --log-level INFO reveals many messages like this:

    The server said My Pokestop was out of spinning range. 13.7m 6.5KMH

At least half of my workers' spin attempts are failing with `OUT_OF_RANGE`, even though Monocle thinks they're well within the current 40m interaction range limit.

Reducing the distance check in `spin_pokestop()` from 36 to a lower value somewhat reduces the failures, but it seems suboptimal to skip pokestops that are within range if some of them would succeed, and it doesn't entirely fix the failures anyway.

Perhaps it would make sense to leave the `next_spin` time untouched on `OUT_OF_RANGE` responses, allowing the worker to try a different pokestop right away instead of having to wait for `SPIN_COOLDOWN` again? It would be helpful to know whether `OUT_OF_RANGE` spin attempts count toward the game's spinning limiter.

**Comment by DavePlater** [comment]
Is niantic checking on altitudes for distances? You are no longer the leg but the hypotenuse of the triangle?
Monocole thinks you're close enough via leg, but the game verifies with the hypotenuse? (If that makes sense?)

**Comment by mewio** [comment]
Making spin_pokestop() sleep for a couple of seconds before it calls fort_details() seems to help significantly. Perhaps the server is doing some kind of movement smoothing / prediction / restriction, and doesn't believe the client's reported location until it has been about the same for a little while?

**Comment by captainmedic** [comment]
I agree, I did a random sleep between 2-5 seconds before spin attempt and the majority succeed now.

**Comment by mewio** [comment]
I wonder if this is related to the times when the game won't let me spin when I have just walked up to a pokestop, even though I wasn't walking quickly. Perhaps the server's speed calculation is bugged.

**Comment by evenly-epic-mule** [comment]
current develop branch:
```
Spun: 39, out of range: 61
```

added `await self.random_sleep(2.0, 5.0)`
```
Spun: 74, out of range: 27
```

so, yes that is a point

**Comment by captainmedic** [comment]
With sleep set random between 4-5.5 I have a spin success rate of ~73.2% on an average over the last hour. Just throwing in my tests of this

**Comment by mewio** [comment]
Copying relevant conversation from PR #293:

captainmedic commented:
> Ok, so I do agree reducing the distance from a pokestop a worker is does indeed increase the passing percentage, currently at 15 minutes of scanning, accounts have spun roughly 800 stops, and have an average success of 79%. I am going to test out a couple more settings, and see how this changes.

evenly-epic-mule commented:
> I've been preparing something simular, which should be a more generic solution

captainmedic commented:
> Just wanted to share another brief experiment with the numbers.
> 
> 82% pass with 377 total attempts to spin - distance 26
> 79.9% 303 total attempts - distance 26
> 91.3% with 173 total attempts to spin - distance 16
> 90.7% with 226 total attempts to spin - - distance 16
> Speed for all tests were set at random 1.8->2.1

evenly-epic-mule commented:
> well I think one main issue here is how far was the last location away
> so we may receive very different results as sizes of maps vary

**Comment by mewio** [comment]
Also worth noting: Percentage of successful spins does not tell the whole story. (One spin in an hour can give 100% by that metric.) Number of successful spins per hour is probably a better measure.

**Comment by mewio** [comment]
This would be an interesting experiment:

* Keep the pokestop range the same as it is in-game.
* Make Monocle workers spin the nearest pokestop when several are in range.
* Make Monocle workers wait a variable amount of time before spinning, depending on how fast they traveled to get there and how far away they are from the pokestop.

**Comment by mewio** [comment]
Unfortunately, I don't have time to spend on this right now, since working on issue #214 is my priority.

**Comment by mewio** [comment]
Another experiment:
* Instead of waiting, make workers use that time to move closer to the pokestop.
